aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -267,6 +267,42 @@ std::tuple<Tensor, Tensor> cummax(const Tensor& self, int64_t dim) {
   at::cummax_out(values, indices, self, dim);
   return std::tuple<Tensor &,Tensor &>{values, indices};
 }
+
+std::tuple<Tensor&, Tensor&> cummin_out(Tensor& values, Tensor& indices, const Tensor& self, int64_t dim) {
+  check_scalar_type_device_layout_equal(values, self);
+  check_scalar_type_device_layout_equal(indices, at::empty({0}, self.options().dtype(at::kLong)));
+  {
+    NoNamesGuard guard;
+    values.resize_(self.sizes());
+    indices.resize_(self.sizes());
+    if(self.dim() == 0) {
+      values.fill_(self.item());
+      indices.fill_(0);
+    }
+    else if(self.numel() != 0) {
+      // update values and indices for the first values along the dimension dim
+      values.narrow(dim, 0, 1) = self.narrow(dim, 0, 1);
```

Comment:
did you benchmark this?  I feel like just using a simple (dataptr-based) algorithm is going to be faster because of the number of tensors you are creating.  How does this compare to, say, min?